src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcBuilderExtensions.cs

```diff
@@ -53,13 +54,17 @@ public static class MvcRazorMvcBuilderExtensions
 
             return builder;
         }
-        
+
         public static IMvcBuilder AddPrecompiledRazorViews(
             [NotNull] this IMvcBuilder builder,
             [NotNull] params Assembly[] assemblies)
         {
-            var razorFileInfos = RazorFileInfoCollections.GetFileInfoCollections(assemblies);
-            builder.Services.TryAddEnumerable(ServiceDescriptor.Instance(razorFileInfos));
+            builder.Services.Replace(
+                ServiceDescriptor.Singleton<ICompilerCacheProvider>(serviceProvider =>
+                    ActivatorUtilities.CreateInstance<PrecompiledViewsCompilerCacheProvider>(
+                        serviceProvider,
+                        assemblies.AsEnumerable())));
```

Comment:
VS ends up greying out the (`IEnumerable<Assembly>`) cast to indicate it is redundant. Rather than have to document why it's there, I'll leave the `AsEnumerable` which essentially does the same thing internally:
http://referencesource.microsoft.com/#System.Core/System/Linq/Enumerable.cs,825